src/box/lua/net_box.lua

```diff
@@ -462,13 +499,26 @@ local function check_eval_args(args)
     end
 end
 
+local function check_request_opts(remote, opts)
+    if opts and opts.is_async then
+        if opts.on_push or opts.on_push_ctx then
+            error('To handle pushes in an async request use future:pairs()')
+        end
+        return
+    end
```

Comment:
I agree with you, there's no need for calling `check_param_table` in each method individually as it can be placed into `_request`. And I've done that.

However, `_request_no_opts_check` (now it has another name) is still needed because we want `_request` to throw an error only in case of connection related problems (let's say, these are the ones, which are returned from `transport`), which is needed for `ping()` not to check the type or message of returned error.  

If we place any checks (including checking the fiber) right inside `_request`, `ping()` have to manually determine whether it should throw an error from `_request` or just return false to the user, if there's an exception inside `_request`. So, I suppose that the best solution is just to put all checks in a separate function. This makes code much cleaner than in case of manual checking of returned error. Just the name of this 'separate function' (check_request_opts) isn't right, `precheck_request` seems to be more appropriate.

Reply:
I think that we can assume that if `ping` is called synchronously from a net.box connection trigger callback, it's a connection-related problem. Anyway,
1. AFAICS there's no test checking that `ping` throws an error in this case.
2. Please don't mix argument checks with this 'Sync requests not allowed` check in the same function.

Reply:
Agree. However, the goal of this check is to avoid a hang when a synchronous request is called from a trigger (by buggy application code). Returning false from `ping()` should be okay in this case. If anybody complains, we might reconsider.